### PR TITLE
Add analytics to VS Code sidebar panel actions

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Work towards https://github.com/flutter/devtools/issues/5868

@DanTup

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/analytics/constants.dart [hunk 1]
```diff
@@ -308,3 +308,20 @@ enum MemoryAreas {
 
   final String name;
 }
+
+enum VsCodeFlutterSidebar {
+  /// Analytics id to track events that come from the Flutter sidebar in VS
+  /// code.
```
Comment (DanTup) [inline packages/devtools_app/lib/src/shared/analytics/constants.dart:314]: uber-nit: "Code" with a capital C :-)

**Review by DanTup — APPROVED** [review]
LGTM!

File: packages/devtools_app/lib/src/shared/analytics/constants.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/analytics/constants.dart:314]: And may be to be on one line?

**Review by polina-c — APPROVED** [review]

**Comment by kenzieschmoll** [comment]
@DanTup please take a look at the query string value we discussed, and let me know what you think.

It should be one of 
`ideFeature=command` // When a user opens DevTools from the command palette 
`ideFeature=sidebarContent` // When a user opens DevTools from the debug session drop down
`ideFeature=sidebarToolbar` // When a user opens DevTools from the action in the sidebar title / tool bar

And this should be passed from VS Code along with the existing (and unchanged) 'ide' query parameter

**Comment by DanTup** [comment]
@kenzieschmoll lgtm - though there are a few additional values you might get because there are quite a few other places we can launch DevTools for (and since I was wiring up these values, it made sense to give them their own).

```
	static commandPalette = "command";
	static sidebarContent = "sidebarContent";
	static sidebarTitle = "sidebarToolbar";
	static touchbar = "touchbar"; // MacOS touchbar button
	static launchConfiguration = "launchConfiguration"; // Configured explicitly in launch configuration
	static onDebugAutomatic = "onDebugAutomatic"; // Configured to always run on debug session start
	static onDebugPrompt = "onDebugPrompt"; // Responded to prompt when running a debug session
	static languageStatus = "languageStatus"; // Launched from the language status popout
```

I suspect the extra 5 are used very little. If having them will cause you any issues, let me know.

image

**Comment by kenzieschmoll** [comment]
Added the values you suggested, @DanTup PTAL

**Comment by DanTup** [comment]
@kenzieschmoll lgtm, though it may be worth noting that that's a non-exhaustive list so if we add additional areas in VS Code we don't necessarily need to keep this list updated?

**Comment by kenzieschmoll** [comment]
added a comment

**Review by DanTup — APPROVED** [review]
Still lgtm :)

**Pull request merged**